components/pedigree/resources/src/main/resources/pedigree/view/datepicker.js

```diff
@@ -1,276 +1,437 @@
-define([
-    "pedigree/model/helpers"
-  ], function(
-    Helpers
-  ){
-
-  var PedigreeFuzzyDatePickerDropdown = Class.create({
-    initialize : function(options) {
-      this.span     = new Element('span');
-      this.options  = options;
-      this.callback = null;
+define([], function(){
+
+  /**
+  * AbstractFuzzyDateInput is an object representing a text input for the @class FuzzyDatePicker
+  *
+  * @class AbstractFuzzyDateInput
+  * @constructor
+  * @param {element} A text input element. The behaviour of the element will be modified.
+  * @param {alwayEnabled} Booolean indicating whether this input should be enabled at all times
+  */
+  var AbstractFuzzyDateInput = Class.create({
+    initialize : function(element, alwaysEnabled){
+      this.element = element;
+      this.alwaysEnabled = alwaysEnabled;
+
+      //The isFuzzy boolean specifies whether the value contained in the iput is specific or fuzzy(a range, decade etc.). Blank and invalid values are not fuzzy.
+      this.isFuzzy = false;
+      this.isValid = true;
+
+      if (this.alwaysEnabled) {
+        this.enable();
+      } else {
+        this.disable();
+      }
+      this.element.onkeypress = this.onKeyPress.bind(this);
+      this.element.observe("focus", this.onFocus.bind(this));
+      this.element.observe("blur", this.onBlur.bind(this));
+      this.element.observe("keydown", this.onKeyDown.bind(this));
+    },
+    getValue : function() {
+      return this.element.value;
+    },
+    isEmpty : function() {
+      return this.getValue() === "";
+    },
+    setValue : function (value) {
+      value && this.enable();
```

Comment:
Fwiw, I find both versions almost equally readable, and I prefer Matt's because it's shorter (which is also why it's so commonly used in JS).